components/team_controller/team_controller.tsx

```diff
@@ -0,0 +1,254 @@
+
+    useEffect(() => {
+        if (props.graphQLEnabled) {
+            props.fetchChannelsAndMembers();
+        } else {
+            props.fetchAllMyTeamsChannelsAndChannelMembersREST();
+        }
+    }, [props.graphQLEnabled]);
+
+    useEffect(() => {
+        const wakeUpIntervalId = setInterval(() => {
+            const currentTime = Date.now();
+            if ((currentTime - lastTime.current) > WAKEUP_THRESHOLD) {
+                console.log('computer woke up - fetching latest'); //eslint-disable-line no-console
+                reconnect(false);
+            }
+            lastTime.current = currentTime;
+        }, WAKEUP_CHECK_INTERVAL);
+
+        return () => {
+            clearInterval(wakeUpIntervalId);
+        };
+    }, []);
+
+    // Effect runs on mount, add event listeners on windows object
+    useEffect(() => {
+        function handleFocus() {
+            if (props.selectedThreadId) {
+                window.isActive = true;
+            }
+            if (props.currentChannelId) {
+                window.isActive = true;
+                props.markChannelAsReadOnFocus(props.currentChannelId);
+            }
+
+            const currentTime = Date.now();
+            if ((currentTime - blurTime.current) > UNREAD_CHECK_TIME_MILLISECONDS && props.currentTeamId) {
+                if (props.graphQLEnabled) {
+                    props.fetchChannelsAndMembers(props.currentTeamId);
+                } else {
+                    props.fetchMyChannelsAndMembersREST(props.currentTeamId);
+                }
+            }
+        }
+
+        function handleBlur() {
+            window.isActive = false;
+            blurTime.current = Date.now();
+
+            if (props.currentUser) {
+                props.viewChannel('');
+            }
+        }
+
+        function handleKeydown(event: KeyboardEvent) {
+            if (event.shiftKey && cmdOrCtrlPressed(event) && isKeyPressed(event, Constants.KeyCodes.L)) {
+                const replyTextbox = document.querySelector<HTMLElement>('#sidebar-right.is-open.expanded #reply_textbox');
+                if (replyTextbox) {
+                    replyTextbox.focus();
+                } else {
+                    const postTextbox = document.getElementById('post_textbox');
+                    if (postTextbox) {
+                        postTextbox.focus();
+                    }
+                }
+            }
+        }
+
+        window.addEventListener('focus', handleFocus);
+        window.addEventListener('blur', handleBlur);
+        window.addEventListener('keydown', handleKeydown);
+
+        return () => {
+            window.removeEventListener('focus', handleFocus);
+            window.removeEventListener('blur', handleBlur);
+            window.removeEventListener('keydown', handleKeydown);
+        };
+    }, [props.selectedThreadId, props.graphQLEnabled, props.currentChannelId, props.currentTeamId, props.currentUser.id]);
```

Comment:
Just sanity check: we won't expect the actions in the props (`markChannelAsReadOnFocus`, `fetchChannelsAndMembers` and `fechMyChannelsAndMembersRest`) to change during the lifetime of the component. Am I right?

Reply:
yes